### PR TITLE
FIR: Added reserved memory area and turned on HugeTLB and dev/mem

### PR DESCRIPTION
**Pull request by atrivedi-tsavoritesi**
1. Turn on /dev/mem access
2. Enabled reserved memory location 0x7F00_0000 to 0x7F1F_FFFF
3. Enabled Huge TLB configuration

### CODE REVIEW

File: arch/arm64/boot/dts/intel/socfpga_agilex_bittware.dts [hunk 1]
```diff
@@ -41,6 +41,34 @@
...
+          };
+
+	/* current allocation is 2 MB from 0x7F00_0000 to 0x7F1F_FFFF
+	*  these allocaiton will change when we add more TXEs and standardize on 
+	*  the space every TXE needs
+	*/
+          display_reserved: framebuffer@7F000000 {
+              reg = <0x7F000000 0x100000>;
+          };
+
```
Comment (akapoor3518) [inline arch/arm64/boot/dts/intel/socfpga_agilex_bittware.dts:65]: please add also comment how much memory is reserved. I will approve

**Review by akapoor3518 — APPROVED** [review]
lgtm

**Review by akapoor3518 — APPROVED** [review]
lgtm

File: arch/arm64/boot/dts/intel/socfpga_agilex_bittware.dts (hunk 1 above)
Comment (atrivedi-tsavoritesi) [reply arch/arm64/boot/dts/intel/socfpga_agilex_bittware.dts:65]: See comment above this line 
/* current allocation is 2 MB from 0x7F00_0000 to 0x7F1F_FFFF
+	*  these allocaiton will change when we add more TXEs and standardize on 
+	*  the space every TXE needs
+	*/

File: arch/arm64/boot/dts/intel/socfpga_agilex_bittware.dts [hunk 2]
```diff
@@ -41,6 +41,34 @@
 		/* We expect the bootloader to fill in the reg */
 		reg = <0 0 0 0>;
 	};
+
```
Comment (tsavoritesi-vgarg) [inline arch/arm64/boot/dts/intel/socfpga_agilex_bittware.dts:44]: Ashish,  Just FYI, Dushmanta is also enabling CMA using same mechanism.  Please check and ensure that there is no conflicts.  Also,  please note  take a note of issues highlighted by NXP on CMA - https://community.nxp.com/t5/i-MX-Processors-Knowledge-Base/How-to-get-rid-of-CMA/ta-p/1123287

**Review by tsavoritesi-vgarg — COMMENTED** [review]
Overall, it look good.  Please review the  issues with NXP highlighted by NXP.

**Review by mmankal — APPROVED** [review]

File: arch/arm64/boot/dts/intel/socfpga_agilex_bittware.dts (hunk 2 above)
Comment (atrivedi-tsavoritesi) [reply arch/arm64/boot/dts/intel/socfpga_agilex_bittware.dts:44]: Will do, I looked up this description and we can revisit after FPGA demo.

**Pull request merged**